src/lib.rs

```diff
@@ -183,6 +192,24 @@ pub trait ToPublicKey: MiniscriptKey {
     /// Converts an object to a public key
     fn to_public_key(&self) -> bitcoin::PublicKey;
 
+    /// Convert an object to x-only pubkey
+    fn to_x_only_pubkey(&self) -> bitcoin::schnorr::PublicKey {
+        let pk = self.to_public_key();
+        bitcoin::schnorr::PublicKey::from(pk.key)
+    }
+
+    /// Serialize the key as bytes based on script context. Used when encoding miniscript into bitcoin script
+    // / - Why do we need both [`MiniscriptKey::is_xonly()`] and [`ScriptContext::Tap`]?
+    // /   Is it possible to have
+    // Named as `serialize_bytes` instead of `serialize` because we already have many functions named serialize
+    fn serialize_bytes<Ctx: ScriptContext>(&self) -> Vec<u8> {
```

Comment:
I would suggest a method like

```
fn push_to_builder<Ctx: ScriptContext>(&self, script::Builder) -> script::Builder
```

which would avoid an allocation and also avoid overloading the name `serialize` any more.